### PR TITLE
add block, transaction, header serialization

### PR DESCRIPTION
**Pull request by bogdan34**

### CODE REVIEW

File: core/driver/socket/index.ts [hunk 1]
```diff
@@ -82,17 +83,29 @@ export class Socket {
...
 
-    onHeadersReceive(response: string, peerAddress: PeerAddress, socket, type: PEER_SOCKET_TYPE) {
-        const peerHeaders = JSON.parse(response);
-        logger.debug(`HEADERS FROM ${type} ${peerAddress.ip}, broadhash: ${peerHeaders.broadhash}, ` +
-            `height: ${peerHeaders.height}`);
+    onHeadersReceive(response: Buffer | string, peerAddress: PeerAddress, socket, type: PEER_SOCKET_TYPE) {
+
+         // TODO delete crutch after migration
+        let peerHeaders;
+        if (Buffer.isBuffer(response)) {
```
Comment (macaqueee) [inline core/driver/socket/index.ts]: how about to extract this code to some `utils` method?

File: shared/repository/transaction/asset/signature.ts [hunk 2]
```diff
@@ -10,6 +12,10 @@ class TransactionSignatureRepo implements IAssetRepository<IAssetSignature> {
         };
     }
 
+    byteSerialize(asset: IAssetSignature): Buffer {
```
Comment (macaqueee) [inline shared/repository/transaction/asset/signature.ts]: we need to refactor this design. serialization/deserialization should not be placed in Repository level

File: core/controller/transaction.ts [hunk 3]
```diff
@@ -13,32 +13,53 @@ import { ResponseEntity } from 'shared/model/response';
 import { CreateTransactionParams } from 'core/controller/types';
 import config from 'shared/config';
 import { ActionTypes } from 'core/util/actionTypes';
+import { VersionChecker } from 'core/util/versionChecker';
+import { BlockModel } from 'shared/model/block';
+import { PeerAddress } from 'shared/model/types';
+import PeerMemoryRepository from 'core/repository/peer/peerMemory';
+
+const BYTE_SEND_VERSION = '1.2.6';
```
Comment (axelsheva) [inline core/controller/transaction.ts]: Maybe is it must be single constant?

File: core/controller/block.ts [hunk 4]
```diff
@@ -14,18 +14,39 @@ import { ActionTypes } from 'core/util/actionTypes';
...
+import { PeerAddress } from 'shared/model/types';
+import PeerMemoryRepository from 'core/repository/peer/peerMemory';
 
 interface BlockGenerateRequest {
     keyPair: IKeyPair;
     timestamp: number;
 }
 
+type BlockReceiveData = {
+    data: BlockModel,
```
Comment (axelsheva) [inline core/controller/block.ts]: This `data` can be renamed as a `blockSchema` or `blockModel`. I think we can to name variable as `data` if we can't definitely name it.

File: shared/repository/transaction/index.ts [hunk 5]
```diff
@@ -51,6 +57,30 @@ class SharedTransactionRepo {
         };
     }
 
+    byteSerialize(trs: Transaction<IAsset>): Buffer {
+        const assetRepo: IAssetRepository<IAsset> = ASSET_REPOSITORIES[trs.type];
+        if (!assetRepo) {
+            logger.error(`[Repository][SharedTransactionRepo][byteSerialize]: unknown transaction type ${trs.type}`);
```
Comment (axelsheva) [inline shared/repository/transaction/index.ts]: Need return error after that?

**Pull request merged**